tests/src/main/resources/sparkconf/spark-defaults.conf

```diff
@@ -0,0 +1,50 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Default system properties included when running spark-submit.
+# This is useful for setting default environmental settings.
+
+spark.default.parallelism           200
+spark.driver.memoryOverhead         1G
+spark.driver.memory                 1G
+spark.executor.memory               8G
+spark.master                        k8s://https://c105.us-east.containers.cloud.ibm.com:32327
```

Comment:
I think you should have `spark.executor.memoryOverhead` set here as well. Also, I believe Kwas was able to reduce executor memory to 4G in his testing after memoryOverhead was increased. It isn't blocking at 8G, but it does use resources unnecessarily for a bit.